datasets/discovery/discovery.py

```diff
@@ -0,0 +1,352 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors and the HuggingFace Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Discourse marker prediction with 174 different markers"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+import textwrap
+
+import six
+
+import datasets
+
+
+_Discovery_CITATION = """@inproceedings{sileo-etal-2019-mining,
+    title = "Mining Discourse Markers for Unsupervised Sentence Representation Learning",
+    author = "Sileo, Damien  and
+      Van De Cruys, Tim  and
+      Pradel, Camille  and
+      Muller, Philippe",
+    booktitle = "Proceedings of the 2019 Conference of the North {A}merican Chapter of the Association for Computational Linguistics: Human Language Technologies, Volume 1 (Long and Short Papers)",
+    month = jun,
+    year = "2019",
+    address = "Minneapolis, Minnesota",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/N19-1351",
+    pages = "3477--3486",
+    abstract = "Current state of the art systems in NLP heavily rely on manually annotated datasets, which are expensive to construct. Very little work adequately exploits unannotated data {--} such as discourse markers between sentences {--} mainly because of data sparseness and ineffective extraction methods. In the present work, we propose a method to automatically discover sentence pairs with relevant discourse markers, and apply it to massive amounts of data. Our resulting dataset contains 174 discourse markers with at least 10k examples each, even for rare markers such as {``}coincidentally{''} or {``}amazingly{''}. We use the resulting data as supervision for learning transferable sentence embeddings. In addition, we show that even though sentence representation learning through prediction of discourse marker yields state of the art results across different transfer tasks, it{'}s not clear that our models made use of the semantic relation between sentences, thus leaving room for further improvements.",
+}
+"""
+
+_Discovery_DESCRIPTION = """\
+Discourse marker prediction with 174 different markers
+"""
+
+DATA_URL = "https://www.dropbox.com/s/aox84z90nyyuikz/discovery.zip?dl=1"
```

Comment:
where does this url come from ?

Reply:
also is there a webpage associated with the dataset somewhere ?

Reply:
> where does this url come from ?

I'm the author of that dataset, I'll add my github link